packages/ozone/src/api/admin/emitModerationEvent.ts

```diff
@@ -2,156 +2,204 @@ import { AuthRequiredError, InvalidRequestError } from '@atproto/xrpc-server'
-        ) {
-          throw new AuthRequiredError(
-            'Must be a full admin to take this type of action on feed generators',
-          )
-        }
-      }
-      // if less than moderator access then can not apply labels
-      if (!access.isModerator && isLabelEvent) {
-        throw new AuthRequiredError('Must be a full moderator to label content')
+      if (status?.takendown && isTakedownEvent) {
+        throw new InvalidRequestError(`Subject is already taken down`)
       }
 
-      if (isLabelEvent) {
-        validateLabels([
-          ...(event.createLabelVals ?? []),
-          ...(event.negateLabelVals ?? []),
-        ])
+      if (!status?.takendown && isReverseTakedownEvent) {
+        throw new InvalidRequestError(`Subject is not taken down`)
       }
 
-      if (isTakedownEvent || isReverseTakedownEvent) {
-        const status = await moderationService.getStatus(subject)
-
-        if (status?.takendown && isTakedownEvent) {
-          throw new InvalidRequestError(`Subject is already taken down`)
-        }
-
-        if (!status?.takendown && isReverseTakedownEvent) {
-          throw new InvalidRequestError(`Subject is not taken down`)
-        }
+      if (status?.takendown && isReverseTakedownEvent && subject.isRecord()) {
+        // due to the way blob status is modeled, we should reverse takedown on all
+        // blobs for the record being restored, which aren't taken down on another record.
+        subject.blobCids = status.blobCids ?? []
+      }
+    }
 
-        if (status?.takendown && isReverseTakedownEvent && subject.isRecord()) {
-          // due to the way blob status is modeled, we should reverse takedown on all
-          // blobs for the record being restored, which aren't taken down on another record.
-          subject.blobCids = status.blobCids ?? []
-        }
+    if (isModEventEmail(event) && event.content) {
+      // sending email prior to logging the event to avoid a long transaction below
+      if (!subject.isRepo()) {
+        throw new InvalidRequestError(
+          'Email can only be sent to a repo subject',
+        )
       }
+      const { content, subjectLine } = event
+      await retryHttp(() =>
+        ctx.modService(db).sendEmail({
+          subject: subjectLine,
+          content,
+          recipientDid: subject.did,
+        }),
+      )
+    }
 
-      if (isModEventEmail(event) && event.content) {
-        // sending email prior to logging the event to avoid a long transaction below
-        if (!subject.isRepo()) {
-          throw new InvalidRequestError(
-            'Email can only be sent to a repo subject',
-          )
-        }
-        const { content, subjectLine } = event
-        await retryHttp(() =>
-          ctx.modService(db).sendEmail({
-            subject: subjectLine,
-            content,
-            recipientDid: subject.did,
-          }),
+    if (isModEventDivert(event) && subject.isRecord()) {
```

Comment:
sync divert blob sequence here before logging the event.
this way, the moderator will immediately know that the diversion failed and the event won't be logged and won't be takendown.